packages/cadl-ranch-specs/http/type/property/optionality/main.tsp

```diff
@@ -237,3 +237,339 @@ interface RequiredAndOptional {
+  @doc("Optional nullable string list")
+  nonRequiredNullableStringList?: string[] | null;
+}
+
+@doc("RoundTrip model with optional properties.")
+model RoundTripOptionalModel {
+  @doc("Optional string, illustrating an optional reference type property.")
+  optionalString?: string;
+
+  @doc("Optional int, illustrating an optional value type property.")
+  optionalInt?: int32;
+
+  @doc("Optional string collection.")
+  optionalStringList?: string[];
+
+  @doc("Optional int collection.")
+  optionalIntList?: int32[];
+
+  @doc("Optional model collection")
+  optionalModelList?: CollectionItem[];
+
+  @doc("Optional model.")
+  optionalModel?: DerivedModel;
+
+  @doc("Optional model with properties on base")
+  optionalModelWithPropertiesOnBase?: DerivedModelWithProperties;
+
+  @doc("Optional fixed string enum")
+  optionalFixedStringEnum?: FixedStringEnum;
+
+  @doc("Optional extensible enum")
+  optionalExtensibleEnum?: ExtensibleEnum;
+
+  @doc("Optional int record")
+  optionalIntRecord?: Record<int32>;
+
+  @doc("Optional string record")
+  optionalStringRecord?: Record<string>;
+
+  @doc("Optional model record")
+  optionalModelRecord?: Record<RecordItem>;
+
+  @doc("Optional plainDate")
+  optionalPlainDate?: plainDate;
+
+  @doc("Optional plainTime")
+  optionalPlainTime?: plainTime;
+
+  @doc("Optional collection of which the element is a nullable int")
+  optionalCollectionWithNullableIntElement?: (int32 | null)[];
+}
+
+@scenario
+@scenarioDoc("""
+Expected response body:
+```json
+{ requiredString: "test",
+  requiredInt: 2,
+  requiredNullableString: null,
+  requiredNullableInt: null,
+  requiredModel: { requiredList: [null] },
+  requiredModel2: { requiredList: [null] },
+  requiredIntList: [1, 2],
+  requiredStringList: ["a", null],
+  requiredModelList: [{ requiredModelRecord: {} }],
+  requiredModelRecord: {},
+  requiredCollectionWithNullableFloatElement: [],
+  requiredCollectionWithNullableBooleanElement: [],
+  requiredNullableModelList: null,
+  requiredNullableStringList: null,
+  requiredNullableIntList: null,
+  }
+```
+
+Expected request body:
+```json
+  {requiredString: "test",
+   requiredInt: 2,
+   requiredNullableString: null,
+   requiredNullableInt: null,
+   requiredReadonlyInt: 3,
+   requiredFixedStringEnum: "1",
+   requiredFixedIntEnum: 2,
+   requiredExtensibleEnum: "1",
+   requiredList: [{ requiredModelRecord: {} }],
+   requiredIntRecord: { "1": 1 },
+   requiredStringRecord: { "1": "1" },
+   requiredModelRecord: {},
+   requiredBytes: "aGVsbG8=",
+   requiredUint8Array: [1, 2],
+   requiredUnknown: "unknown",
+   requiredInt8Array: [1, 2],
+   requiredNullableIntList: [1, 2, null],
+   requiredNullableStringList: ["a", "b", null],
+   nonRequiredNullableIntList: [1, 2, null],
+   nonRequiredNullableStringList: ["a", "b", null],
+    }
+   }
+```
+""")
+@route("/inputToRoundTrip")
+@doc("Input to RoundTrip")
+@get
+@convenientAPI(true)
+op inputToRoundTrip(@body input: InputModel): RoundTripModel;
```

Comment:
This is the only place I see `RoundTripModel`, then why it is a roundtrip?

Reply:
This part is entirely moved from Models_TypeSpec. The RoundTripModel can serve as either input or output. In Models, the operations for input and output are separate. Here, the migration is done in reference to the inputToRoundTripReadOnly section.